tools/util/include/cutlass/util/GPU_Clock.hpp

```diff
@@ -47,10 +47,16 @@ struct GPU_Clock
   }
 
   void start() {
+#if defined(CUTLASS_ENABLE_SYCL)
+    syclcompat::get_default_queue().wait();
+#endif
     cudaEventRecord(start_);
   }
 
   float milliseconds() {
+#if defined(CUTLASS_ENABLE_SYCL)
+    syclcompat::get_default_queue().wait();
```

Comment:
Why is a device level synchronization required before event synchronization ?

Reply:
Because these are cuda events

Reply:
I see, in that case, instead of synchronizing the entire device, let's wait on the event returned by the `syclcompat::launch`, similar to [this](https://github.com/AD2605/cutlass-fork/blob/48c40846b48c2fbb6a0b41bc1aaf84879d1caa1c/examples/cute/tutorial/sgemm_1_sycl.cpp#L407), and as an added benefit, this timer struct will remain unchanged

Reply:
The call to `syclcompat::launch` is inside Cutlass. We don't have access to that from the cutlass API.